platform/jewel/markdown/core/src/main/kotlin/org/jetbrains/jewel/markdown/MarkdownMode.kt

```diff
@@ -0,0 +1,29 @@
+// Copyright 2000-2025 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package org.jetbrains.jewel.markdown
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import org.jetbrains.jewel.foundation.ExperimentalJewelApi
+import org.jetbrains.jewel.markdown.extensions.LocalMarkdownMode
+import org.jetbrains.jewel.markdown.scrolling.ScrollingSynchronizer
+
+@ExperimentalJewelApi
+public sealed interface MarkdownMode {
+    public val withEditor: Boolean
+    public val scrollingSynchronizer: ScrollingSynchronizer?
```

Comment:
I'd remove it from here and leave it only in WithEditor, since it's always null in the WithPreview mode